Цель: Реализовать базовый анализ датасета с помощью функций библеотекы Pandas и Jupiter

Ход работы:

Создание датасета из CSV файла

In [2]:
import pandas as pd
df = pd.read_csv("C:\\Users\\User\\Desktop\\N\\VS _ VS Code\\Project PY\\anime.csv")

Вывод датасета

In [3]:
display(df)

anime_id                                               name  \
0         32281                                     Kimi no Na wa.   
1          5114                   Fullmetal Alchemist: Brotherhood   
2         28977                                           Gintama°   
3          9253                                        Steins;Gate   
4          9969                                      Gintama&#039;   
...         ...                                                ...   
12289      9316       Toushindai My Lover: Minami tai Mecha-Minami   
12290      5543                                        Under World   
12291      5621                     Violence Gekiga David no Hoshi   
12292      6133  Violence Gekiga Shin David no Hoshi: Inma Dens...   
12293     26081                   Yasuji no Pornorama: Yacchimae!!   

                                                   genre   type episodes  \
0                   Drama, Romance, School, Supernatural  Movie        1   
1      Action, Adventure, Drama, Fantasy, Magic, Mili...     TV       64   
2      Action, Comedy, Historical, Parody, Samurai, S...     TV       51   
3                                       Sci-Fi, Thriller     TV       24   
4      Action, Comedy, Historical, Parody, Samurai, S...     TV       51   
...                                                  ...    ...      ...   
12289                                             Hentai    OVA        1   
12290                                             Hentai    OVA        1   
12291                                             Hentai    OVA        4   
12292                                             Hentai    OVA        1   
12293                                             Hentai  Movie        1   

       rating  members  
0        9.37   200630  
1        9.26   793665  
2        9.25   114262  
3        9.17   673572  
4        9.16   151266  
...       ...      ...  
12289    4.15      211  
12290    4.28      183  
12291    4.88      219  
12292    4.98      175  
12293    5.46      142  

[12294 rows x 7 columns]

Мой датасет сожержит информацию об аниме различных жанров и типов, к которой относится, например, количество зрителей и рейтинг

Вывод размероности

In [4]:
display(df.shape)

(12294, 7)

Размерность отображает количество строк и столбцов в датасете. В моём датасете 12294 и 7 соответственно.

Вывод наименования всех колонок

In [5]:
display(df.columns)

Index(['anime_id', 'name', 'genre', 'type', 'episodes', 'rating', 'members'], dtype='object')

В моём датасете есть 7 столбцов 
Anime id - идентификационный номер
Name - название аниме
Genre - жанр аниме
Type - тип аниме
Episodes - количество эпизодов
Rating - рейтинг аниме 
Members - количество просмотров 

Описание колонок

Сортировка по ID

In [6]:
sorted= df.sort_values(by='anime_id')
display(sorted)

anime_id                                               name  \
22            1                                       Cowboy Bebop   
152           5                    Cowboy Bebop: Tengoku no Tobira   
214           6                                             Trigun   
2095          7                                 Witch Hunter Robin   
3159          8                             Beet the Vandel Buster   
...         ...                                                ...   
2726      34514                                Pokemon Generations   
9586      34519                Mobile Suit Gakuen: G-Reco Koushien   
11104     34522                          Wake Up, Girls! Shin Shou   
10952     34525                                  Centaur no Nayami   
10997     34527  Gou-chan. Moko to Chinjuu no Mori no Nakama-tachi   

                                                   genre     type episodes  \
22       Action, Adventure, Comedy, Drama, Sci-Fi, Space       TV       26   
152                Action, Drama, Mystery, Sci-Fi, Space    Movie        1   
214                               Action, Comedy, Sci-Fi       TV       26   
2095   Action, Drama, Magic, Mystery, Police, Superna...       TV       26   
3159           Adventure, Fantasy, Shounen, Supernatural       TV       52   
...                                                  ...      ...      ...   
2726              Action, Adventure, Fantasy, Game, Kids      ONA       18   
9586                                              Comedy  Special        9   
11104                                       Drama, Music       TV  Unknown   
10952       Comedy, Fantasy, Slice of Life, Supernatural       TV  Unknown   
10997                                    Adventure, Kids    Movie        1   

       rating  members  
22       8.82   486824  
152      8.40   137636  
214      8.32   283069  
2095     7.36    64905  
3159     7.06     9848  
...       ...      ...  
2726     7.21      295  
9586     5.67       94  
11104     NaN      381  
10952     NaN      108  
10997     NaN        5  

[12294 rows x 7 columns]

Датасет был отсортирован по ID в порядке возрастания , что может быть полезным для выявления незанятых ID

Удаление столбца "members"

In [7]:
df = df.drop(columns=['members'])
display(df.columns)

Index(['anime_id', 'name', 'genre', 'type', 'episodes', 'rating'], dtype='object')

Замена количества эпизодов с 1 на 7559

In [9]:
df = df.replace({"episodes":"1"},7559)
display(df['episodes'])

0        7559
1          64
2          51
3          24
4          51
         ... 
12289    7559
12290    7559
12291       4
12292    7559
12293    7559
Name: episodes, Length: 12294, dtype: object

В данном действии я заменил все значения эпизодов, которые были равны 1 на 7559

Удаление дубликатов

In [10]:
df = df.drop_duplicates()
display(df)

anime_id                                               name  \
0         32281                                     Kimi no Na wa.   
1          5114                   Fullmetal Alchemist: Brotherhood   
2         28977                                           Gintama°   
3          9253                                        Steins;Gate   
4          9969                                      Gintama&#039;   
...         ...                                                ...   
12289      9316       Toushindai My Lover: Minami tai Mecha-Minami   
12290      5543                                        Under World   
12291      5621                     Violence Gekiga David no Hoshi   
12292      6133  Violence Gekiga Shin David no Hoshi: Inma Dens...   
12293     26081                   Yasuji no Pornorama: Yacchimae!!   

                                                   genre   type episodes  \
0                   Drama, Romance, School, Supernatural  Movie     7559   
1      Action, Adventure, Drama, Fantasy, Magic, Mili...     TV       64   
2      Action, Comedy, Historical, Parody, Samurai, S...     TV       51   
3                                       Sci-Fi, Thriller     TV       24   
4      Action, Comedy, Historical, Parody, Samurai, S...     TV       51   
...                                                  ...    ...      ...   
12289                                             Hentai    OVA     7559   
12290                                             Hentai    OVA     7559   
12291                                             Hentai    OVA        4   
12292                                             Hentai    OVA     7559   
12293                                             Hentai  Movie     7559   

       rating  
0        9.37  
1        9.26  
2        9.25  
3        9.17  
4        9.16  
...       ...  
12289    4.15  
12290    4.28  
12291    4.88  
12292    4.98  
12293    5.46  

[12294 rows x 6 columns]

Анализи датасета с помощью функций info

In [11]:
display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12294 entries, 0 to 12293
Data columns (total 6 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   anime_id  12294 non-null  int64  
 1   name      12294 non-null  object 
 2   genre     12232 non-null  object 
 3   type      12269 non-null  object 
 4   episodes  12294 non-null  object 
 5   rating    12064 non-null  float64
dtypes: float64(1), int64(1), object(4)
memory usage: 576.4+ KB


None

1. RangeIndex сообщает нам, что у нас есть 12294 записи (или строк) в нашем DataFrame, индексируемых от 0 до 12293.

2. Data columns содержит информацию о каждом столбце:
   - anime_id: содержит 12294 непустых значений типа int64.
   - name: содержит 12294 непустых значений типа object (текстовая строка).
   - genre: содержит 12232 непустых значений типа object.
   - type: содержит 12269 непустых значений типа object.
   - episodes: содержит 12294 непустых значений типа object.
   - rating: содержит 12064 непустых значений типа float64.

3. Dtype сообщает нам о типах данных в каждом столбце. У нас есть один столбец типа int64, один столбец типа float64 и четыре столбца типа object.

4. Memory usage (использование памяти) показывает, сколько памяти занимает наш DataFrame, которая составляет 576.4+ KB.


Анализи датасета с помощью функций describe

In [12]:
display(df.describe())

anime_id        rating
count  12294.000000  12064.000000
mean   14058.221653      6.473902
std    11455.294701      1.026746
min        1.000000      1.670000
25%     3484.250000      5.880000
50%    10260.500000      6.570000
75%    24794.500000      7.180000
max    34527.000000     10.000000

1. count показывает количество значений в каждом столбце. Видно, что у нас есть 12294 значений в anime_id и 12064 в rating, что означает, что есть некоторые пропущенные значения в столбце rating.

2. mean - это среднее значение. Для anime_id это около 14058, а для rating около 6.47. Это дает общее представление о средних значениях в наших данных.

3. std - это стандартное отклонение, которое показывает, насколько сильно значения разбросаны вокруг среднего значения. Для anime_id это около 11455, а для rating около 1.03.

4. min и max - минимальное и максимальное значения в каждом столбце. Минимальное значение для anime_id - 1, а для rating - 1.67. Максимальное значение для anime_id - 34527, а для rating - 10.

5. 25%, 50%, и 75% - это квартили, они показывают, как данные разделены на четыре равные части. Например, 25% наблюдений в столбце anime_id имеют значение, меньшее или равное 3484.25, а 50% имеют значение, меньшее или равное 10260.5.

Выборку данных по строкам и столбцам с помощью loc для аниме с названием Код Гиас :Восставший Лелуш

In [28]:
selected_data = df.loc[df['name'] == 'Code Geass: Hangyaku no Lelouch']
display(selected_data)

anime_id                             name  \
19      1575  Code Geass: Hangyaku no Lelouch   

                                                genre type episodes  rating  
19  Action, Mecha, Military, School, Sci-Fi, Super...   TV       25    8.83

Сохранение нового датасета

In [31]:
df.to_csv('anime_sorted.csv', index=False)

Вывод: В ходе работы был проанализирован датасет с аниме, также в него были внесены некоторые изменения. В процессе работы были изучены базовые функции библеотеки Pandas 